curvefs/src/metaserver/s3/metaserver_s3_adaptor.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 2021-8-13
+ * Author: chengyi
+ */
+
+#include "metaserver_s3_adaptor.h"
+
+namespace curvefs {
+namespace metaserver {
+void S3ClientAdaptorImpl::Init(const S3ClientAdaptorOption &option,
+                               S3Client *client) {
+    blockSize_ = option.blockSize;
+    chunkSize_ = option.chunkSize;
+    client_ = client;
+}
+
+int S3ClientAdaptorImpl::Delete(Inode *inode) {
+    S3ChunkInfoList s3ChunkInfolist(inode->s3chunkinfolist());
```

Comment:
1. I copy it for that: the inode deleted by others during delete s3 objects;
2. who call this fuction, and it will delete inode so inode->s3chunkinfolist() will be deleted by destructor。

Reply:
No need to  considering  "deleted by others during delete s3 objects"